tools/ninja/patches/100-make_jobserver_support.patch

```diff
@@ -1,58 +1,99 @@
-From 17d13fd7881fd3ce9f9b9d44ce435d6caf4b8f28 Mon Sep 17 00:00:00 2001
-From: Stefan Becker <stefanb@gpartner-nvidia.com>
-Date: Tue, 22 Mar 2016 13:48:07 +0200
-Subject: [PATCH 01/11] Add GNU make jobserver client support
-
-- add new TokenPool interface
-- GNU make implementation for TokenPool parses and verifies the magic
-  information from the MAKEFLAGS environment variable
-- RealCommandRunner tries to acquire TokenPool
-  * if no token pool is available then there is no change in behaviour
-- When a token pool is available then RealCommandRunner behaviour
-  changes as follows
-  * CanRunMore() only returns true if TokenPool::Acquire() returns true
-  * StartCommand() calls TokenPool::Reserve()
-  * WaitForCommand() calls TokenPool::Release()
-
-Documentation for GNU make jobserver
-
-  http://make.mad-scientist.net/papers/jobserver-implementation/
-
-Fixes https://github.com/ninja-build/ninja/issues/1139
----
- configure.py              |   2 +
- src/build.cc              |  63 ++++++++----
- src/build.h               |   3 +
- src/tokenpool-gnu-make.cc | 211 ++++++++++++++++++++++++++++++++++++++
- src/tokenpool-none.cc     |  27 +++++
- src/tokenpool.h           |  26 +++++
- 6 files changed, 310 insertions(+), 22 deletions(-)
- create mode 100644 src/tokenpool-gnu-make.cc
- create mode 100644 src/tokenpool-none.cc
- create mode 100644 src/tokenpool.h
-
-diff --git a/configure.py b/configure.py
-index 43904349a8..db3492c93c 100755
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
```

Comment:
@hauke this patchset was added at @nbd 's request. https://github.com/ninja-build/ninja/pull/1140

Interestingly enough, KitWare (makers of CMake) have their own fork of Ninja with just this patchset applied. Might make sense to switch tools/ninja to that.

Reply:
Probably. I would also prefer doing that.

Reply:
@hauke denied this. Might as well set this to ready.

Reply:
I would prefer if we continue using the original repository.
The https://github.com/Kitware/ninja repository says this:
> This branch may be rebased without notice for maintenance on top of the upstream master branch. It will be removed once upstream has integrated the features.

I do not like that. 

Could you please update the patch with the code from the https://github.com/ninja-build/ninja/pull/1140 pull request or the https://github.com/Kitware/ninja repository and name where you got it from in your commit message.

Reply:
Done. I am right now very busy. :/ Could someone give it a test?